### ISSUE TITLE
Record.formatJSON(Writer) and Record.formatXML(Writer) should flush the writer

### ISSUE THREAD

**Issue by kairlec** [jOOQ/jOOQ#13023]
### Expected behavior
write json string onto outputstream
### Actual behavior
Leaving at most 8192 bytes in the StreamEncoder's Buffer
### Steps to reproduce the problem
```java
ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
anyRecord.formatJSON(byteArrayOutputStream);
log.info("Writing record to json {}", byteArrayOutputStream);
```
If the json length is less than 8193, what I get will be empty string.
I noticed the code in this place: [AbstractFormattable.java](https://github.com/jOOQ/jOOQ/blob/ccad787ebc1dd5a162c73be26e4579f024b7d5ce/jOOQ/src/main/java/org/jooq/impl/AbstractFormattable.java#L264-L272). Is it because forgot to add a call to `flush` or `close` or is there something wrong with my writing?

### Versions

- jOOQ: 3.16.4
- Java: openjdk 11.0.13
- Database (include vendor):
- OS:
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce similar problems when calling `Record.formatJSON(OutputStream)`, though not when calling `Result.formatJSON(OutputStream)`.

Indeed, the `Record` version is missing a `Writer::flush` call. Will check if that alone fixes it.

**Comment by lukaseder** [comment]
`Record::formatXML` is also affected

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13025), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13026).

Thanks again for your report.
